ganga/GangaCore/Utility/feedback_report.py

```diff
@@ -72,57 +71,47 @@ def get_content_type(filename):
         
         return data, file_dict
 
-    def make_upload_file(server):
-
-        def upload_file(path):
-
-            # print 'Uploading %r to %r' % (path, server)
-
-            data = {'MAX_FILE_SIZE': '3145728',
-                    'sub': '',
-                    'mode': 'regist'}
-            files = {'file': path}
-
-            return send_post(server, files)
-
-        return upload_file
+    def upload(filetype, filename, localdir):
+        try:
+                from GangaCore.GPIDev.Adapters.IGangaFile import IGangaFile
+                from GangaCore.GPIDev.Lib.File.LocalFile import LocalFile
+                from GangaCore.GPIDev.Lib.File.LCGSEFile import LCGSEFile
+                from GangaCore.GPIDev.Lib.File.GoogleFile import GoogleFile
+                from GangaCore.GPIDev.Lib.File.MassStorageFile import MassStorageFile
+                
+                filetype = stripProxy(filetype)
+                assert(issubclass(filetype, IGangaFile))
+                print(filename, localdir)
+                if issubclass(filetype, LocalFile):
+                    feedback = filetype(namePattern = filename)
+                    feedback.localDir = localdir 
 
-    def send_post(url, files):
-        logger.debug("Sending Post to %s ,  containing %s" % (url, files))
+                else:
+                    feedback = filetype(filename)
+                    feedback.localDir = localdir 
+                    feedback.put()
 
-        encoded_data = encode_multipart_formdata(files)
+        except Exception as e:
+                raise GangaFileError(f"filetype {filetype} is not supported")
         
-        #response = requests.post(url, data=encoded_data[0], files=encoded_data[1])
-        response = send_request(url,encoded_data[0],encoded_data[1])
+        if issubclass(filetype, GoogleFile):
+            logger.info(
+                f'Your error report was uploaded as {feedback.downloadURL}.')  
+            logger.info(
+                'You may include this URL and the following summary information in your bug report or in the support email to the Ganga developers.')
+        else:
+            logger.info(
+                f'Your error report was saved as {filetype.location}.')
+            logger.info(
+                'You may include this file and the following summary information in your bug report or in the support email to the Ganga developers.')
```

Comment:
Avoid this if/else again. Even for LocalFile, there should be a URL (file:// ....).

Reply:
All `Files` have an `accessUrl` while `GoogleFile` has a `downloadUrl` attribute. This means we would require an `if` condition to cater for this.

Reply:
Yes, please rename in `GoogleFile`.